packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -523,6 +540,12 @@ class MediaQuery extends InheritedWidget {
     return MediaQuery.of(context, nullOk: true)?.textScaleFactor ?? 1.0;
   }
 
+  /// Returns platformBrightness for the nearest MediaQuery ancestor or
+  /// [PlatformBrightness.no_preference], if no such ancestor exists.
```

Comment:
It would be worth pointing out that context will be rebuilt if any aspect of the MediaQuery changes

Reply:
Done